File: python/LeetCode/03289.py
from typing import List

def getSneakyNumbers(self, nums: List[int]) -> List[int]:
    n = len(nums)
    seen = set()

    output = []
    for i in nums:
        if i in seen:
            output.append(i)
            if len(output) == 2:
                return output
        seen.add(i)

File: go/leetcode/03289.go
package main

func getSneakyNumbers(nums []int) []int {
	seen := make(map[int]struct{})

	output := make([]int, 0, 2)

	for _, num := range nums {
		if _, ok := seen[num]; ok {
			output = append(output, num)
			if len(output) == 2 {
				return output
			}
		}
		seen[num] = struct{}{}
	}

	return output
}
